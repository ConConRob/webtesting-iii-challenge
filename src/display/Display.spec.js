// Test away!
import React from "react";
import * as rt from "react-testing-library";
import Display from "./Display";

afterEach(rt.cleanup);

describe("Display", () => {
  it("Closed if closed prop is true", () => {
    const wrap = rt.render(<Display closed={true} />);
    expect(wrap.getByText(/closed/i));
    expect(wrap.asFragment()).toMatchSnapshot();
  });
  it("Closed with red-led if closed prop is true", () => {
    const wrap = rt.render(<Display closed={true} />);
    expect(wrap.queryByText(/closed/i).classList.contains("red-led")).toBe(
      true
    );
  });
  it("Open if closed prop is false", () => {
    const wrap = rt.render(<Display closed={false} />);
    expect(wrap.getByText(/open/i));
    expect(wrap.asFragment()).toMatchSnapshot();
  });
  it("Open with red-led if closed prop is false", () => {
    const wrap = rt.render(<Display closed={false} />);
    expect(wrap.queryByText(/open/i).classList.contains("green-led")).toBe(
      true
    );
  });
  it("locked if locked prop is true", () => {
    const wrap = rt.render(<Display locked={true} />);
    expect(wrap.getByText(/locked/i));
    expect(wrap.asFragment()).toMatchSnapshot();
  });
  it("locked with red-led if locked prop is true", () => {
    const wrap = rt.render(<Display locked={true} />);
    expect(wrap.queryByText(/locked/i).classList.contains("red-led")).toBe(
      true
    );
  });
  it("unlocked if locked prop is false", () => {
    const wrap = rt.render(<Display locked={false} />);
    expect(wrap.getByText(/unlocked/i));
    expect(wrap.asFragment()).toMatchSnapshot();
  });
  it("unlocked with green-led if locked prop is false", () => {
    const wrap = rt.render(<Display locked={false} />);
    expect(wrap.queryByText(/unlocked/i).classList.contains("green-led")).toBe(
      true
    );
  });
});
