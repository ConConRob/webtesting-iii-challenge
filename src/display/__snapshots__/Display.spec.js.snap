// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Closed if closed prop is true 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
</DocumentFragment>
`;

exports[`Display Open if closed prop is false 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;

exports[`Display locked if locked prop is true 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led red-led"
    >
      Locked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;

exports[`Display unlocked if locked prop is false 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</DocumentFragment>
`;
